src/Jackalope/Transport/DoctrineDBAL/Query/QOMWalker.php

```diff
@@ -690,12 +690,7 @@ public function walkNumComparisonConstraint(QOM\PropertyValueInterface $property
             );
         }
 
-        return sprintf(
-            '%s %s %s',
-            $this->sqlXpathExtractNumValue($alias, $property),
-            $operator,
-            $literalOperand->getLiteralValue()
-        );
+        return $this->sqlXpathComparePropertyValue($alias, $property, $literalOperand->getLiteralValue(), $operator);
```

Comment:
Also not sure how this is working with other impls. as the logic this replaces casts the integer. Perhaps this should be done only for Sqlite